Formatting GoDoc special comment blocks

Daniel-M opened the pull request:
I've limited the length of all the special comment block on the `.go` files. I've used an `awk` script to do it, checked all the special comments character count and none of them gave a length higher than 78 characters including the `//` chars.

mrd0ll4r commented on api.go:
```diff
@@ -1,5 +1,7 @@
-// Copyright 2015-2016 mrd0ll4r and contributors. All rights reserved.
-// Use of this source code is governed by the MIT license, which can be found in
+// Copyright 2015-2016 mrd0ll4r and contributors. All rights 
```
The copyright header is not godoc and does not need to be 80 chars wide

mrd0ll4r commented on api.go:
```diff
@@ -36,16 +38,20 @@ func (u *BotUpdate) Update() Update {
 }
 
-// Error returns != nil, if an error occurred during retrieval of the update
+// Error returns != nil, if an error occurred during retrieval of 
```
This was originally 77 chars wide, does not need a line break

mrd0ll4r commented:
I think something about your script is not working correctly - see the comment I marked that originally had a length of 77 chars.

While you're at it, you could also make sure that all godoc comments are in the proper format (we need correct punctuation):
https://blog.golang.org/godoc-documenting-go-code

Daniel-M commented:
Restructured comment blocks with an improved script, Now it passes travis tests.

mrd0ll4r commented on api.go:
```diff
@@ -173,9 +179,11 @@ func (api *TelegramBotAPI) GetMe() (*UserResponse, error) {
+// download a file.
 // You will have to construct the download link manually like
-// https://api.telegram.org/file/bot<token>/<file_path>, where <file_path> is taken from the response.
+// https:// api.telegram.org/file/bot<token>/<file_path>, where <file_path>
```
There was a space added to the URI

mrd0ll4r commented on doc.go:
```diff
@@ -2,15 +2,20 @@
 //
-// We currently only support long polling (i.e. no webhooks). Feature-wise, everything up to and including the January 20
+// We currently only support long polling (i.e. no webhooks). Feature-wise,
+// everything up to and including the January 20
```
pull the next line up

mrd0ll4r commented on doc.go:
```diff
@@ -2,15 +2,20 @@
-// The Bot API imposes certain limitations, these are especially interesting for inline query results and files. This
-// library does not keep track of those limitations, so you'll have to perform checks yourself.
+// The Bot API imposes certain limitations, these are especially
+// interesting for inline query results and files. This
```
pull the next line up

Daniel-M commented:
What I did,   
- Joined lines you've pointed out.  
- Added punctuation to documentation comments requiring it.  
- Fixed the URIs splitted by script. (The ones you've pointed and others).  
- Reformated non-documentation comments so the format looks more uniform.

mrd0ll4r commented on examples/boilerplate/boilerplate.go:
```diff
@@ -1,6 +1,6 @@
 // Copyright 2015-2016 mrd0ll4r and contributors. All rights reserved.
 // Use of this source code is governed by the MIT license, which can be found in
-// the LICENSE file.
+// The LICENSE file.
```
This doesn't need to be capitalized.

mrd0ll4r commented on examples/boilerplate/boilerplate.go:
```diff
@@ -15,12 +15,13 @@ import (
 // THIS IS JUST FOR DEMONSTRATION! NOT TO BE USED IN PRODUCTION!
-// It will block until either something very bad happens or closing is closed.
+// It will block until either something very bad happens or closing is
+// Closed.
```
This doesn't need to be capitalized either.

mrd0ll4r commented on examples/keyboardMarkup.go:
```diff
@@ -21,29 +21,29 @@ func main() {
-			// msg.Chat implements fmt.Stringer, so it'll display nicely
-			// MessageType implements fmt.Stringer, so it'll display nicely
+			// Display the incoming message.
+			// Msg.Chat implements fmt.Stringer, so it'll display nicely.
```
Don't capitalize here, `msg` refers to the variable.

mrd0ll4r commented on examples/markup.go:
```diff
@@ -20,18 +20,18 @@ func main() {
-			// msg.Chat implements fmt.Stringer, so it'll display nicely
-			// we know it's a text message, so we can safely use the Message.Text pointer
+			// Display the incoming message.
+			// Msg.Chat implements fmt.Stringer, so it'll display nicely.
```
Same here.

mrd0ll4r commented on examples/photo.go:
```diff
@@ -21,25 +21,25 @@ func main() {
-			// msg.Chat implements fmt.Stringer, so it'll display nicely
-			// we know it's a text message, so we can safely use the Message.Text pointer
+			// Display the incoming message.
+			// Msg.Chat implements fmt.Stringer, so it'll display nicely.
```
Same here.

mrd0ll4r commented on incoming.go:
```diff
@@ -333,15 +336,16 @@ type PhotoSize struct {
 }
 
-// Sticker represents a sticker
+// Sticker represents a sticke.
```
This is missing an r.

mrd0ll4r commented on incoming.go:
```diff
@@ -333,15 +336,16 @@ type PhotoSize struct {
 
-// UpdateResponse represents the response sent by the API for a GetUpdates request
+// UpdateResponse represents the response sent by the API for a GetUpdate
+// request
```
This needs a full stop at the end.

mrd0ll4r commented on incoming.go:
```diff
@@ -433,7 +437,8 @@ func (u User) String() string {
 
-// UserProfilePhotosResponse represents the response sent by the API on a GetUserProfilePhotos request
+// UserProfilePhotosResponse represents the response sent by the API on a
+// GetUserProfilePhotos request
```
This needs a full stop.

mrd0ll4r commented on outgoing.go:
```diff
@@ -436,15 +449,18 @@ type InlineKeyboardButton struct {
-// ForceReply represents the values sent by a bot so that clients will be presented with a forced reply, see https://core.telegram.org/bots/api#forcereply
+// ForceReply represents the values sent by a bot so that clients will be
+// presented with a forced reply,
+// see https://core.telegram.org/bots/api#forcereply
```
You can pull some of this up.

mrd0ll4r commented on doc.go:
```diff
@@ -2,15 +2,18 @@
+// in a Go error.
 //
 // We currently only support long polling (i.e. no webhooks). Feature-wise, everything up to and including the January 20
 // changes should be implemented.
```
reformat these two lines

mrd0ll4r commented on doc.go:
```diff
@@ -2,15 +2,18 @@
 //
 // Examples are provided in the examples package, so check that out.
 //
 // The Bot API imposes certain limitations, these are especially interesting for inline query results and files. This
```
pull up the rest of then sentence, then reformat this line

mrd0ll4r commented on examples/chatAction.go:
```diff
@@ -1,6 +1,6 @@
 // Copyright 2015-2016 mrd0ll4r and contributors. All rights reserved.
 // Use of this source code is governed by the MIT license, which can be found in
-// the LICENSE file.
+// The LICENSE file.
```
don't change the license header.

mrd0ll4r commented on examples/echo.go:
```diff
@@ -1,6 +1,6 @@
 // Copyright 2015-2016 mrd0ll4r and contributors. All rights reserved.
 // Use of this source code is governed by the MIT license, which can be found in
-// the LICENSE file.
+// The LICENSE file.
```
don't change the license header

mrd0ll4r commented:
LGTM
